runhouse/servers/http/http_utils.py

```diff
@@ -281,3 +301,143 @@ def handle_response(
     elif output_type == OutputType.STDERR:
         res = response_data["data"]
         print(system_color + res + reset_color, file=sys.stderr)
+
+
+###########################
+#### Folder Operations ####
+###########################
+def folder_mkdir(path: Path):
+    if not path.parent.is_dir():
+        raise ValueError(
+            f"Parent path {path.parent} does not exist or is not a directory"
+        )
+
+    path.mkdir(parents=True, exist_ok=True)
+
+    return Response(output_type=OutputType.SUCCESS)
+
+
+def folder_get(path: Path, folder_params: FolderParams):
+    mode = folder_params.mode or "rb"
+    serialization = folder_params.serialization
+    binary_mode = "b" in mode
+
+    with open(path, mode=mode) as f:
+        file_contents = f.read()
+
+    if binary_mode and isinstance(file_contents, bytes):
+        file_contents = file_contents.decode()
+
+    output_type = OutputType.RESULT_SERIALIZED
+    serialization = serialization or ("pickle" if binary_mode else None)
+
+    return Response(
+        data=file_contents,
+        output_type=output_type,
+        serialization=serialization,
+    )
+
+
+def folder_put(path: Path, folder_params: FolderParams):
+    overwrite = folder_params.overwrite
+    mode = folder_params.mode or "wb"
+    serialization = folder_params.serialization
+    contents = folder_params.contents
+
+    if not path.is_dir():
+        raise ValueError(f"Path {path} is not a directory")
+
+    path.mkdir(exist_ok=True)
+
+    if overwrite is False:
+        existing_files = {str(item.name) for item in path.iterdir()}
+        intersection = existing_files.intersection(set(contents.keys()))
+        if intersection:
+            raise FileExistsError(
+                f"File(s) {intersection} already exist(s) at path: {path}"
+            )
+
+    for filename, file_obj in contents.items():
+        binary_mode = "b" in mode
```

Comment:
@dongreenberg marking this now to discuss later, the server serialization / deserialization business for file contents getting and putting feels a little hairy